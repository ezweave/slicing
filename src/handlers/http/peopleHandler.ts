import { APIGatewayProxyHandler } from 'aws-lambda';
import { StatusCodes } from 'http-status-codes';

import { GetNPeople, getNPeople } from '@slicing/operation';
import { getPathParamFromAPIGatewayProxyEvent } from '@slicing/utilities/aws';

export const handler =
  (getNPeopleFn: GetNPeople = getNPeople): APIGatewayProxyHandler =>
  async (event, _context) => {
    const n = getPathParamFromAPIGatewayProxyEvent('n', parseInt)(event);
    const people = await getNPeopleFn(n);
    return {
      statusCode: StatusCodes.OK,
      body: JSON.stringify(people),
    };
  };

export const peopleHandler = handler();
