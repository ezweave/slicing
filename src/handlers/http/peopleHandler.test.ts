import { APIGatewayProxyEvent, Context } from 'aws-lambda';
import { peopleHandler } from './peopleHandler';

const event = {} as unknown as APIGatewayProxyEvent;
const context = {} as unknown as Context;
const callback = jest.fn();

describe(peopleHandler, () => {
  it('returns a response', async () => {
    expect(await peopleHandler(event, context, callback)).toMatchSnapshot();
  });
});
