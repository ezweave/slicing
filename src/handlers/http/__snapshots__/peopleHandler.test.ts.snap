// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` returns a response 1`] = `
Object {
  "body": "[]",
  "statusCode": 200,
}
`;
